docs/user_guide/mitigation.rst

```diff
@@ -80,11 +81,408 @@ Reductions
+
+.. note::
+
+    It is not possible to specify both :code:`difference_bound` *and*
+    :code:`ratio_bound` for the same constraint object.
+
+.. _demographic_parity:
+
+Demographic Parity
+^^^^^^^^^^^^^^^^^^
+
+A classifier :math:`h(X)` satisfies *Demographic Parity* if
+
+.. math::
+    
+    \P[h(X) = y' \given A = a] = \P[h(X) = y'] \; \forall a, y'
+ 
+In other words, the selection rate or percentage of samples with label 1
+should be equal between all groups. Implicitly this means the percentage
+with label 0 is equal as well.
+
+In the example below class :code:`"a"` has a selection rate of 60%,
+:code:`"b"` has a selection rate of 20%. The overall selection rate is 40%,
+so :code:`"a"` is `0.2` above the overall selection rate, and :code:`"b"` is
+`0.2` below. The seemingly redundantly stored constraint violation with both
+:code:`"a"` and :code:`"b"` showing up twice stems from the constraint
+definition outlined in :ref:`constraints_binary_classification`. Each equality
+constraint is written as two inequalities of which one contains
+:math:`\text{metric}_a` positively (sign :math:`+`) and one negatively
+(sign :math:`-`).
+Note that the specified :code:`difference_bound` is not used
+since the constraint object is not used in a mitigation in this example.
+Similarly, :code:`y_true` is technically irrelevant to the calculations
+because the underlying metric of Demographic Parity, selection rate, does not
+consider performance relative to the true labels, but rather proportions in
+the predicted labels.
+
+.. note::
+
+    In mitigation examples we never need to call :code:`load_data` or
+    :code:`gamma` since the reductions techniques all automatically take care
+    of it. The example merely helps in illustrating the deviation from the
+    corresponding inequalities in :ref:`constraints_binary_classification`.
+
+.. doctest:: mitigation
+
+    >>> from fairlearn.reductions import DemographicParity
+    >>> from fairlearn.metrics import selection_rate_group_summary
+    >>> import numpy as np
+    >>> dp = DemographicParity(difference_bound=0.01)
+    >>> X                  = np.array([[0], [1], [2], [3], [4], [5], [6], [7], [8], [9]])
+    >>> y_true             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  0 ,  0 ,  0 ])
+    >>> y_pred             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  0 ,  0 ,  0 ])
+    >>> sensitive_features = np.array(["a", "b", "a", "a", "b", "a", "b", "b", "a", "b"])
+    >>> selection_rate_group_summary(y_true, y_pred, sensitive_features=sensitive_features)
+    {'overall': 0.4, 'by_group': {'a': 0.6, 'b': 0.2}}
+    >>> dp.load_data(X, y_true, sensitive_features=sensitive_features)
+    >>> dp.gamma(lambda X: y_pred)
+    sign  event  group_id
+    +     all    a           0.2
+                 b          -0.2
+    -     all    a          -0.2
+                 b           0.2
+    dtype: float64
+ 
+Rewriting this into a ratio-based constraint with :code:`ratio_bound`
```

Comment:
"Alternatively, we can write this as a ratio-based constraint"

I don't believe that it will give the same ultimate answers will it? In which case, it's not really 'rewriting' I think.

Reply:
Of course the answers are different, that's the entire point of showing the example twice :-)

> "To rewrite" = to make a revision of (something, such as a story)

We are rewriting it by changing the argument.

Reply:
To me, 'rewriting' an equation implies that it's the same one, rearranged a bit. I don't believe that's the case for difference vs ratio bounds

Reply:
I think Miro's revisions removed this.